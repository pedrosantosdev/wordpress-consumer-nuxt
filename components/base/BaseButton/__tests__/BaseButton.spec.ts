import { describe, it, expect, afterEach } from 'vitest'
import { fireEvent, render } from '@testing-library/vue'
import BaseButtonVue from '@/components/base/BaseButton/BaseButton.vue'

const dataPropsRender = [
	{
		label: 'Icon and Disabled',
		props: {
			disabled: true,
			showIcon: true,
		},
	},
	{
		label: 'Icon and not Disabled',
		props: {
			disabled: false,
			showIcon: true,
		},
	},
	{
		label: 'Text and Disabled',
		props: {
			disabled: true,
			showIcon: false,
		},
	},
	{
		label: 'Text and not Disabled',
		props: {
			disabled: false,
			showIcon: false,
		},
	},
]
const defaultLabelSlot = 'Login'

describe('BaseButtonVue', () => {
	afterEach(() => {
		document.body.innerHTML = ''
	})
	describe.each(dataPropsRender)('should render properly', (data) => {
		it(data.label, () => {
			const { html }= render(BaseButtonVue, {
				props: data.props,
				slots: { default: defaultLabelSlot },
			})
			expect(html()).toMatchSnapshot()
		})
	})
	it(`should emit click properly when not disabled`, async () => {
		const { emitted, getByText } = render(BaseButtonVue, {
			slots: { default: defaultLabelSlot },
		})
		await fireEvent.click(getByText(defaultLabelSlot))
		expect(emitted().click).toBeTruthy()
		expect(emitted().click.length).toBe(1)
	})

	it(`should not emit click when disabled`, async () => {
		const { container, emitted } = render(BaseButtonVue, {
			props: { disabled: true },
			slots: { default: defaultLabelSlot },
		})
		await fireEvent.click(container)
		expect(emitted().click).toBeFalsy()
	})
})
