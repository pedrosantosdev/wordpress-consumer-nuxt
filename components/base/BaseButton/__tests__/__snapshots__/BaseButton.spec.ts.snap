// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButtonVue > should render properly > Icon and Disabled 1`] = `
"<button type=\\"button\\" disabled=\\"\\" class=\\"py-2 px-4 h-10 bg-purple-600 hover:bg-purple-700 focus:ring-purple-500 focus:ring-offset-purple-200 text-white w-full transition ease-in duration-200 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg\\">
  <nuxticon class=\\"spinner\\" name=\\"spinner\\"></nuxticon>
</button>"
`;

exports[`BaseButtonVue > should render properly > Icon and not Disabled 1`] = `
"<button type=\\"button\\" class=\\"py-2 px-4 h-10 bg-purple-600 hover:bg-purple-700 focus:ring-purple-500 focus:ring-offset-purple-200 text-white w-full transition ease-in duration-200 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg\\">
  <nuxticon class=\\"spinner\\" name=\\"spinner\\"></nuxticon>
</button>"
`;

exports[`BaseButtonVue > should render properly > Text and Disabled 1`] = `"<button type=\\"button\\" disabled=\\"\\" class=\\"py-2 px-4 h-10 bg-purple-600 hover:bg-purple-700 focus:ring-purple-500 focus:ring-offset-purple-200 text-white w-full transition ease-in duration-200 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg\\">Login</button>"`;

exports[`BaseButtonVue > should render properly > Text and not Disabled 1`] = `"<button type=\\"button\\" class=\\"py-2 px-4 h-10 bg-purple-600 hover:bg-purple-700 focus:ring-purple-500 focus:ring-offset-purple-200 text-white w-full transition ease-in duration-200 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg\\">Login</button>"`;
